### DISCUSSION TITLE
Managed metadata column does not appear in the Search Refiner webpart (v4)

### DISCUSSION THREAD

**Discussion by faisal-imam** [microsoft-search/pnp-modern-search#1881] [Q&A] [answered]
I’m using PnP Modern Search WebPart v4 – the latest build which is v 4.5.4. 

I’ve created a Content Type using the Content Type Gallery in SharePoint Admin Center – lets call this content type ‘XYZ Document’. This content type has a Metadata column called ‘Business Function’ the values of which are derived from a term set in SharePoint global taxonomy (defined in Admin Center). 
I then associated this ‘XYZ Document’ content type with a bunch of SharePoint team sites’ default document library. Let’s call these sites ‘TeamSiteA’, ‘TeamSiteB’ and ‘TeamSiteC‘. I then added some documents in the ‘Documents’ library for all the three team sites. I then associated all the three team sites to a new SharePoint Communication Hub site – let’s  call that site ‘Documents Hub’.
After doing all these configurations, I dropped the PnP Modern Search Result WebPart on the home page of ‘Documents Hub’. I turned on the ‘Enable Localization’ on this webpart. I also this webpart to retrieves only documents whose content type is ‘XYZ Document’ i.e. in other words, it only brings content from ‘TeamSiteA’, ‘TeamSiteB’ and ‘TeamSiteC’. I have specified the following properties in the ‘Selected properties’ option of the Search Results webpart:
•	File Type
•	owstaxIdBusinessx0020function
•	Path
•	Title
•	UniquieID
Using the Debug view of the Search Results webpart, I can see the value of ‘Business Function’ for each document in the selected properties owstaxIdBusinessx0020function (which shows the termID and value) and (automatically generated property called) AutoowstaxIdBusinessx0020function (which shows just the term value). This is how the Debug view roughly looks like:
 
 
![image](https://user-images.githubusercontent.com/49614150/155874256-0c9c1b85-887e-407d-a362-9990bfb9dbf0.png)
![image](https://user-images.githubusercontent.com/49614150/155874279-3f7b6ceb-e7af-43c6-8430-accb4860c76d.png)
![image](https://user-images.githubusercontent.com/49614150/155874282-208706e7-3275-4224-9b50-7e48b864b3ae.png)
![image](https://user-images.githubusercontent.com/49614150/155874285-ab40d31a-0353-4df9-b405-cd1862382846.png)
![image](https://user-images.githubusercontent.com/49614150/155874286-4fed281b-dce0-4843-a3d8-d1246b173874.png)
![image](https://user-images.githubusercontent.com/49614150/155874290-9d065d20-a144-4063-93d2-6c08827e7816.png)

   
 
 

The problem is that when I configure the PnP Search Fitlers WebPart and add two refiners, one for File Type and another for Business Function, it just doesn’t show the Business Function.
Configuration of Search Filters webpart:

![image](https://user-images.githubusercontent.com/49614150/155874295-9a6540d9-00ac-4042-be8b-afb8cf94b321.png)

 
Output:

![image](https://user-images.githubusercontent.com/49614150/155874300-55b735c7-f163-47ca-b307-306d91b4bbfc.png)

 
As you can see, I am not able to see the Business Function filter in the output. There seems to be some problem showing the managed metadata filters.
I haven’t done any additional configuration in the Managed Properties. This is how the Managed Properties look like:

![image](https://user-images.githubusercontent.com/49614150/155874306-7168bb50-2cbf-42e4-9f25-a447aa3098d1.png)

 
Any idea why the Search Filters webpart is not showing the Business Function?

**Comment by wobba** [comment, marked as answer]
`owstaxIdBusinessx0020function` is not refinable, thus cannot be used as a filter. See **Enable localization** at https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/

**Reply by faisal-imam** [reply]
Good catch @wobba. I later realized that too and later mapped the crawled property Business function to existing managed property RefinableString00. 

![image](https://user-images.githubusercontent.com/49614150/156082026-007ad913-89d2-4666-8bc5-08d951eaa9f7.png)

When I try using this RefinableString00 managed property, the Search Results webpart shows me the value of this property in this format:

![image](https://user-images.githubusercontent.com/49614150/156082060-677149fc-395e-440e-b0f8-0111aa5fad5c.png)

And the Search Filter webpart appears like this when I select the RefinableString00 as a Filter.

![image](https://user-images.githubusercontent.com/49614150/156082102-eb764672-378e-4091-937a-0086172c4a6f.png)

I have Localization already enabled on the Search Results webpart. Any idea how can I get rid of the extra GUIDs appearing in the Search Filters and just show the labels instead.

**Reply by FranckyC** [reply]
@faisal-imam it seems you mapped the incorrect crawled property. The value of `RefinableString00` is not correct. You must map the `ows_taxId_BusinessFunction` to get the localization work (and use the `AutoRefinableString00` as slot value).

**Reply by faisal-imam** [reply]
Thanks a lot @FranckyC . I mapped the `RefinableString00` to `ows_taxId_BusinessFunction`. It works now!

**Reply by stchiew** [reply]
@faisal-imam Did you manage to get the 'Business Function' filter values to show just the labels and not the GUIDs? I followed the steps in this discussion and managed to display correctly in search results using the AutoRefinableString00, but if I use AutoRefinableString00 in the filters, it does not appear. If I use RefinableString00, the filter appears but with the GUIDs as above. What Managed Property should I be using in the filters?

**Reply by faisal-imam** [reply]
@stchiew Yes I did manage to get it working. I'm using RefinableString00 for the filters and it shows me the text value (without GUIDs). Just make sure you re-index the source libraries/site and also make sure the 'Enable localization' is ON in the search results webpart.

**Reply by stchiew** [reply]
@faisal-imam NICE! Yes, the re-indexing of the site did it for me. Thanks for the tips!
